app/models/jurisdiction.rb

```diff
@@ -162,23 +162,32 @@ def zero_carbon_step_required(activity = nil, permit_type = nil)
   def create_integration_mappings
     return unless external_api_enabled?
 
+    existing_mapping_template_ids = integration_mappings.pluck(:template_version_id)
+
     relevant_template_versions =
       TemplateVersion
         .published
         .or(TemplateVersion.deprecated.where(deprecation_reason: "new_publish"))
+        .where.not(id: existing_mapping_template_ids)
         .order(version_date: :asc)
 
-    existing_mapping_template_ids = integration_mappings.pluck(:template_version_id)
-
-    templates_without_mappings = relevant_template_versions.where.not(id: existing_mapping_template_ids)
-
-    templates_without_mappings.each do |template_version|
+    relevant_template_versions.each do |template_version|
       integration_mappings.create(template_version: template_version)
     end
   end
 
   private
 
+  def create_integration_mappings_async
+    return unless external_api_enabled?
+
+    if Rails.env.test?
+      ModelCallbackJob.new.perform(self.class.name, id, "create_integration_mappings")
```

Comment:
are you certain that two ModelCallbackJobs with the same class, id, and callback name will never need to be run at the same time?

Reply:
@bkmaibach thats what the lock is for. It has unique lock based on the params, which in this case would be the model id, model name and callback name. I've configured it so that it rejects any further jobs with same parameters. There is a couple of options for conflict resolution but I thought this made the most sense as we shouldn't need to run the same job with the same params. Please let me know if another default would be preferable